packages/extension/src/lib/extensionScripts.ts

```diff
@@ -1,29 +1,63 @@
-import 'content-scripts-register-polyfill';
-import { browser, ContentScripts } from 'webextension-polyfill-ts';
-import {
-  CreateRequestContentScripts,
-  contentScriptKey,
-} from '@dailydotdev/shared/src/hooks';
+import browser from 'webextension-polyfill';
+import { contentScriptKey } from '@dailydotdev/shared/src/hooks';
 import { companionPermissionGrantedLink } from '@dailydotdev/shared/src/lib/constants';
-import { AnalyticsEvent } from '@dailydotdev/shared/src/lib/analytics';
+import { AnalyticsEvent as AnalyticsEventName } from '@dailydotdev/shared/src/lib/analytics';
+import { QueryClient } from '@tanstack/react-query';
+import { AnalyticsEvent } from '@dailydotdev/shared/src/hooks/analytics/useAnalyticsQueue';
 
-export const registerBrowserContentScripts =
-  (): Promise<ContentScripts.RegisteredContentScript> =>
-    browser.contentScripts.register({
+export type RequestContentScripts = (data: {
+  origin: string;
+  skipRedirect?: boolean;
+}) => Promise<boolean>;
+
+export type CreateRequestContentScripts = (
+  client: QueryClient,
+  trackEvent: (e: AnalyticsEvent) => void,
+) => RequestContentScripts;
+
+export const HOST_PERMISSIONS = [
+  'https://daily.dev/*',
+  'https://*.daily.dev/*',
+];
+
+let hasInjectedScripts = false;
+const companionScriptId = 'daily-companion-app';
+
+export const registerBrowserContentScripts = async (): Promise<void> => {
+  if (hasInjectedScripts) {
+    return null;
+  }
+
+  const registeredScripts = await browser.scripting.getRegisteredContentScripts(
+    {
+      ids: [companionScriptId],
+    },
+  );
```

Comment:
I think this can just always call `browser.scripting.getRegisteredContentScripts` because if for any reason user disallows content script from chrome menu `hasInjectedScripts` will be outdated. I don't think actual call to `getRegisteredContentScripts` will be much of a overhead because it is a Promise to local browser APIs. 

So basically I would remove `hasInjectedScripts` optimization.

Reply:
You can remove `hasInjectedScripts` completely now right, meaning declaration and all assignments?

Reply:
Since we are still using V2 API on FF via `browser.contentScripts.register`, we still need it for now

Reply:
You are right, tnx for clarification 🚀